### ISSUE TITLE
Superset dashboard fails to load for anonymous user

### ISSUE THREAD

**Issue by davemulford** [apache/superset#14504]
Superset dashboard fails to load for an anonymous user (i.e. not logged in). This error does not happen when logged in with the admin user.

### Expected results

The dashboard loads and displays the charts I've created.

### Actual results

The browser window reports an _"Unexpected Error"_ and looking in the developer tools window, the console displays the following error:

```
TypeError: can't convert undefined to object
    $< http://plmdashboards.usersys.redhat.com/static/assets/dashboard.b03ba823dd0eabc8f3a0.entry.js:380
    a http://plmdashboards.usersys.redhat.com/static/assets/dashboard.b03ba823dd0eabc8f3a0.entry.js:55
    ne http://plmdashboards.usersys.redhat.com/static/assets/dashboard.b03ba823dd0eabc8f3a0.entry.js:380
    r http://plmdashboards.usersys.redhat.com/static/assets/dashboard.b03ba823dd0eabc8f3a0.entry.js:432
    _p http://plmdashboards.usersys.redhat.com/static/assets/dashboard.b03ba823dd0eabc8f3a0.entry.js:1754
    ...
```

#### Screenshots

Showing the error in browser (not very helpful):
![image](https://user-images.githubusercontent.com/243049/117327338-1bb2f480-ae58-11eb-84ee-b376057b8f83.png)

Showing the error in the developer console:
![image](https://user-images.githubusercontent.com/243049/117327431-35543c00-ae58-11eb-800e-d0d9def9a578.png)

#### How to reproduce the bug

1. Create a dashboard
2. Copy the dashboard URL to a private browser window, or a different browser that is not logged into Superset.

### Environment

(please complete the following information):

- superset version: `Version: 0.999.0dev`
- python version: `Python 3.7.9`
- node.js version: `node -v`

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

I am using the `docker-compose-non-dev.yml` to run Superset. I have tried various container images (`docker.io/apache/superset:latest` and `docker.io/apache/superset:latest-dev`), as well as, switching to the latest release tag of `v2021.15.0`.

I've added the `PUBLIC_ROLE_LIKE_GAMMA = True` to the `./docker/pythonpath_dev/superset_config.py` file and added the datasource permissions that are used in the dashboard.

**Comment by davemulford** [comment]
Forgot to include the `Public` role permissions, so I've done that below.

```
[can read on SavedQuery, can write on SavedQuery, can read on CssTemplate, can write on CssTemplate,
can read on ReportSchedule, can write on ReportSchedule, can read on Chart, can write on Chart, can read on Annotation,
can write on Annotation, can read on Dataset, can write on Dataset, can read on Dashboard, can write on Dashboard,
can read on Database, can write on Database, can read on Query, can this form get on ResetMyPasswordView,
can this form post on ResetMyPasswordView, can this form get on UserInfoEditView, can this form post on UserInfoEditView,
can userinfo on UserDBModelView, resetmypassword on UserDBModelView, can get on OpenApi, can show on SwaggerView,
can get on MenuApi, can list on AsyncEventsRestApi, can invalidate on CacheRestApi, can time range on Api,
can query form data on Api, can query on Api, can this form get on CsvToDatabaseView,
can this form post on CsvToDatabaseView, can this form get on ExcelToDatabaseView,
can this form post on ExcelToDatabaseView, can external metadata on Datasource, can get on Datasource,
can shortner on R, can my queries on SqlLab, can user slices on Superset, can log on Superset, can fave slices on Superset,
can filter on Superset, can extra table metadata on Superset, can add slices on Superset, can sqllab on Superset,
can sqllab history on Superset, can request access on Superset, can schemas on Superset, can sqllab viz on Superset,
can validate sql json on Superset, can annotation json on Superset, can save dash on Superset,
can available domains on Superset, can dashboard on Superset, can stop query on Superset,
can sqllab table viz on Superset, can profile on Superset, can fetch datasource metadata on Superset,
can estimate query cost on Superset, can slice on Superset, can copy dash on Superset, can queries on Superset,
can created slices on Superset, can results on Superset, can created dashboards on Superset,
can schemas access for csv upload on Superset, can tables on Superset, can publish on Superset,
can testconn on Superset, can recent activity on Superset, can explore json on Superset, can warm up cache on Superset,
can import dashboards on Superset, can explore on Superset, can search queries on Superset, can datasources on Superset,
can slice json on Superset, can csrf token on Superset, can fave dashboards by username on Superset,
can select star on Superset, can favstar on Superset, can fave dashboards on Superset, can csv on Superset,
can post on TableSchemaView, can expanded on TableSchemaView, can delete on TableSchemaView,
can post on TabStateView, can activate on TabStateView, can migrate query on TabStateView, can put on TabStateView,
can delete query on TabStateView, can delete on TabStateView, can get on TabStateView, can read on SecurityRestApi,
menu access on List Users, menu access on List Roles, menu access on Action Log, menu access on Annotation Layers,
menu access on Import Dashboards, menu access on Alerts & Report, menu access on Data, menu access on Databases,
menu access on Datasets, menu access on Upload Excel, menu access on Charts, menu access on Dashboards,
menu access on SQL Editor, menu access on Saved Queries, menu access on Query Search,
all datasource access on all_datasource_access, all database access on all_database_access,
can share dashboard on Superset, can share chart on Superset, database access on [DATABASE_NAME].(id:2),
schema access on [DATABASE_NAME].[TABLE_NAME], datasource access on [DATABASE_NAME].[TABLE_NAME](id:29),
datasource access on [DATABASE_NAME].[TABLE_NAME](id:32)]
```

**Comment by zufolo441** [comment]
We have this problem too. there is not anyone in the log.

<img width="592" alt="Schermata 2021-05-14 alle 06 16 22" src="https://user-images.githubusercontent.com/45826835/118220455-09e6d800-b47c-11eb-8101-d434f19d7521.png">

**Comment by MilenaLee** [comment]
same here on Version: 0.999.0dev
![image](https://user-images.githubusercontent.com/19401376/118257012-1a816780-b4e9-11eb-9953-49b87c231ab8.png)

**Comment by zufolo441** [comment]
the problem is still present: 
Uncaught TypeError: can't convert undefined to object
    isUserAdmin findPermission.ts:39
    canUserEditDashboard findPermission.ts:52
    hydrateDashboard hydrate.js:325
    createThunkMiddleware Redux
    DashboardPage DashboardPage.tsx:70
    commitHookEffectListMount react-dom.development.js:19765
    commitPassiveHookEffects react-dom.development.js:19803
    callCallback react-dom.development.js:189
    invokeGuardedCallbackDev react-dom.development.js:238
    invokeGuardedCallback react-dom.development.js:293
    flushPassiveEffectsImpl react-dom.development.js:22885
    unstable_runWithPriority scheduler.development.js:653
    runWithPriority$1 react-dom.development.js:11062
    flushPassiveEffects react-dom.development.js:22852
    commitBeforeMutationEffects react-dom.development.js:22731
    workLoop scheduler.development.js:597
    flushWork scheduler.development.js:552
    performWorkUntilDeadline scheduler.development.js:164
    EventHandlerNonNull* scheduler.development.js:187
    <anonymous> scheduler.development.js:857
    node_modules vendors.ff448c0c.chunk.js:22005
    __webpack_require__ spa.ff448c0c.entry.js:85
    <anonymous> index.js:6
    node_modules vendors.ff448c0c.chunk.js:22017
    __webpack_require__ spa.ff448c0c.entry.js:85
    <anonymous> react-dom.development.js:21
    <anonymous> react-dom.development.js:25105
    node_modules vendors.ff448c0c.chunk.js:21969
    __webpack_require__ spa.ff448c0c.entry.js:85
    <anonymous> index.js:37
    node_modules vendors.ff448c0c.chunk.js:21981
    __webpack_require__ spa.ff448c0c.entry.js:85
    <anonymous> index.tsx:1
    tsx spa.ff448c0c.entry.js:2539
    __webpack_require__ spa.ff448c0c.entry.js:85
    5 spa.ff448c0c.entry.js:2576
    __webpack_require__ spa.ff448c0c.entry.js:85
    checkDeferredModules spa.ff448c0c.entry.js:46
    <anonymous> spa.ff448c0c.entry.js:250
    <anonymous> spa.ff448c0c.entry.js:253
findPermission.ts:39

**Comment by michael-s-molina** [comment]
@davemulford According to Superset [docs](https://superset.apache.org/docs/security) the property in `superset_config.py` is `PUBLIC_ROLE_LIKE = 'Gamma'` instead of `PUBLIC_ROLE_LIKE_GAMMA = True`. Can you also check that the dashboard you're trying to share is owned by a Gamma user and that the `Public` role has access to the data source being used by the dashboard?

**Comment by albertwgchu** [comment]
@michael-s-molina I am seeing this issue as well.  Confirmed that a Gamma user owns the dashboard and the public role has access to the data.  I have the following in my config:

AUTH_ROLE_PUBLIC = 'Public'
PUBLIC_ROLE_LIKE = 'Gamma'

**Comment by michael-s-molina** [comment]
@suddjian You probably have the complete context here. Can you help us?

**Comment by davemulford** [comment]
@michael-s-molina I've changed to using `PUBLIC_ROLE_LIKE = 'Gamma'` and restarted Superset. I then created a new user and added the Gamma role to that user. I also made sure the newly created Gamma user was an owner of the dashboard.

I still see the same problem when trying to access the dashboard without being logged in. Note that when I log in with the Gamma user, I can view the dashboard. So it seems like the `PUBLIC_ROLE_LIKE = 'Gamma'` doesn't work as expected?
